Lib/logging/__init__.py

```diff
@@ -1558,33 +1563,31 @@ def findCaller(self, stack_info=False, stacklevel=1):
         f = currentframe()
         #On some versions of IronPython, currentframe() returns None if
         #IronPython isn't run with -X:Frames.
-        if f is not None:
-            f = f.f_back
-        orig_f = f
-        while f and stacklevel > 1:
-            f = f.f_back
-            stacklevel -= 1
-        if not f:
-            f = orig_f
-        rv = "(unknown file)", 0, "(unknown function)", None
-        while hasattr(f, "f_code"):
-            co = f.f_code
-            filename = os.path.normcase(co.co_filename)
-            if filename == _srcfile:
-                f = f.f_back
-                continue
-            sinfo = None
-            if stack_info:
-                sio = io.StringIO()
-                sio.write('Stack (most recent call last):\n')
+        if f is None:
+            return "(unknown file)", 0, "(unknown function)", None
+        while stacklevel > 0:
+            next_f = f.f_back
+            if next_f is None:
+                ##TODO: We've got options here
+                ## If we want to use the last (deepest) frame:
+                break
+                ## If we want to mimic the warnings module:
+                #return ("sys", 1, "(unknown function)", None)
+                ## If we want to be pedantic:
+                #raise ValueError("call stack is not deep enough")
```

Comment:
@vsajip Thanks for the approval and merge! Note that there was a TODO here which I summarized in the original PR description as:
 > Regarding the last point, I did not know what to do in case more frames are to be skipped than the height of the stack. I listed three possibilities in the code which I think would all be sensible.

Having it merged like this is not wrong per se (it will use the deepest available frame if the stack is exhausted), but it would be good to at least remove the `TODO:` part on line 1571. Maybe all of the above could even be replaced with
```python
            if next_f is None:
                # Not enough stack frames. We use the last (deepest) one.
                break
```

Reply:
Or maybe leave it as a reminder in case this needs revisiting for any reason?

Reply:
Indeed, but in that case I would still remove the word `TODO:`, since nothing _needs_ to be done.

Anyway, I really appreciate that you took a look at this PR. Thanks!